src/engine/runtime.js

```diff
@@ -650,14 +686,32 @@ class Runtime extends EventEmitter {
         this._pushMonitors();
         const doneThreads = this.sequencer.stepThreads();
         this._updateGlows(doneThreads);
-        this._setThreadCount(this.threads.length + doneThreads.length);
+        // Add done threads so that even if a thread finishes within 1 frame, the green
+        // flag will still indicate that a script ran.
+        this._maybeEmitProjectRunStartOrStop(
+            this.threads.length - this._getMonitorThreadCount(this.threads) +
+                doneThreads.length - this._getMonitorThreadCount(doneThreads));
```

Comment:
Not a big deal but this could just do one function call with `this.threads.length + doneThreads.length - this._getMonitorThreadCount([...this.threads, ...doneThreads])`.